moveit_planners/pilz_industrial_motion_planner/test/python_move_group_planning.py

```diff
@@ -0,0 +1,117 @@
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+# Author: Cristian C. Beltran-Hernandez
+#
+# This test is used to ensure planning with an attached object
+# correctly validates collision states between the attached objects
+# and the environment
+
+import unittest
+import numpy as np
+import rospy
+import rostest
+import os
+
+from moveit_msgs.msg import MoveItErrorCodes
+import moveit_commander
+
+from geometry_msgs.msg import PoseStamped
+
+
+class PythonMoveGroupPlanningTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        PLANNING_GROUP = "panda_arm"
+        self.group = moveit_commander.MoveGroupCommander(PLANNING_GROUP)
+        self.planning_scene_interface = moveit_commander.PlanningSceneInterface(
+            synchronous=True
+        )
+
+    @classmethod
+    def tearDown(self):
+        pass
+
+    def test_planning_with_collision_objects(self):
+        # Add obstacle to the world
+        ps = PoseStamped()
+        ps.header.frame_id = "world"
+        ps.pose.position.x = 0.4
+        ps.pose.position.y = 0.1
+        ps.pose.position.z = 0.25
+        self.planning_scene_interface.add_box(
+            name="box1", pose=ps, size=(0.1, 0.1, 0.5)
+        )
+
+        # Attach object to robot's TCP
+        ps2 = PoseStamped()
+        tcp_link = self.group.get_end_effector_link()
+        ps2.header.frame_id = tcp_link
+        ps2.pose.position.z = 0.15
+        self.planning_scene_interface.attach_box(
+            link=tcp_link,
+            name="box2",
+            pose=ps2,
+            size=(0.1, 0.1, 0.1),
+            touch_links=["panda_rightfinger", "panda_leftfinger"],
+        )
+
+        # Plan a motion where the attached object 'box2' collides with the obstacle 'box1'
+        target_pose = self.group.get_current_pose(tcp_link)
+        target_pose.pose.position.y += 0.1
+
+        # # Set planner to be Pilz's Linear Planner
+        # self.group.set_planning_pipeline_id("pilz_industrial_motion_planner")
+        # self.group.set_planner_id("LIN")
+        # self.group.set_pose_target(target_pose)
+        # success, plan, time, error_code = self.group.plan()
+
+        # # Planning should fail
+        # self.assertEqual(error_code.val, MoveItErrorCodes.INVALID_MOTION_PLAN)
+
+        # Set planner to be Pilz's Point-To-Point Planner
+        self.group.set_planning_pipeline_id("pilz_industrial_motion_planner")
+        self.group.set_planner_id("PTP")
+        self.group.set_pose_target(target_pose)
+        success, plan, time, error_code = self.group.plan()
+
+        # Planning should fail
+        self.assertEqual(error_code.val, MoveItErrorCodes.INVALID_MOTION_PLAN)
```

Comment:
maybe also test the return value for the sake of completeness?
```suggestion
        self.assertFalse(success)
        self.assertEqual(error_code.val, MoveItErrorCodes.INVALID_MOTION_PLAN)
```

Reply:
Sure